### ISSUE TITLE
Show error messages for permissions errors in the compact index code

### ISSUE THREAD

**Issue by lynnco** [bundler/bundler#4647]
So we can get better error messages for issues like https://github.com/bundler/bundler/issues/4590

**Comment by RochesterinNYC** [comment]
This would be great, though not sure how we want to approach this. The compact index code, as an external censored dependency, has no knowledge of Bundler. We could possibly extract the modules related to filesystem access and erroring, but this extracted module of functionality would then become a dependency of both Bundler and the compact index code separately.

We could also move the compact index functionality into Bundler.

**Comment by segiddins** [comment]
@RochesterinNYC let the compact index take a proc property to access the filesystem that by default only does the access, without error handling

**Comment by indirect** [comment]
Another potential solution could be wrapping `Errno::*` errors in a `CompactIndex::AccessError` that has accessors for both the underlying error and the path being accesses, so that Bundler can catch those errors and print out a useful message. It's definitely doable, one way or another.

**Comment by allenzhao** [comment]
Is this one solved by #4795 ?

**Comment by LynnCo** [comment]
@allenzhao No, [indirect's comment](https://github.com/bundler/bundler/issues/4647#issuecomment-223865277) is still the preferred solution. #4795 does make this issue lower priority, though.

**Comment by lynnco** [comment]
Blocked by https://github.com/bundler/bundler/issues/4935

**Comment by segiddins** [comment]
Ah this is actually done in https://github.com/bundler/bundler/pull/5007!
